modules/mono/glue/GodotSharp/GodotSharp/Core/NodePath.cs

```diff
@@ -288,5 +288,37 @@ public bool IsAbsolute()
         /// </summary>
         /// <returns>If the <see cref="NodePath"/> is empty.</returns>
         public bool IsEmpty => NativeValue.DangerousSelfRef.IsEmpty;
+
+        public static bool operator ==(NodePath left, NodePath right)
+        {
+            if (left is null)
+                return right is null;
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(NodePath left, NodePath right)
+        {
+            return !(left == right);
+        }
+
+        public bool Equals(NodePath other)
+        {
+            if (other is null)
+                return false;
+            var self = (godot_node_path)NativeValue;
+            var otherNative = (godot_node_path)other.NativeValue;
+            return NativeFuncs.godotsharp_node_path_equals(self, otherNative).ToBool();
+        }
```

Comment:
This implementation also seems more costly than what we do in `StringName`. Can't we just compare `NativeValue.DangerousSelfRef`?

Reply:
Because in `StringName` the whole point of StringNames is that all equal values are the same pointer and NativeValue, is that pointer, but for `NodePath` no such grantee exists, so we need to compare the pointer targets, which is native data, so we either need to implement the layout of that type or call into native code.